File: pset2/vigenere.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
#include<ctype.h>

bool check(string alph);

int main(int argc, string argv[]){
    if(!(argc==2)  || (check(argv[1])==false)){
        printf("Usage: ./vigenere keyword\n");
        return 1;
    }else{
        int shift[strlen(argv[1])];
        for(int i = 0; i < strlen(argv[1]); i++){
            if(islower(argv[1][i])){
                shift[i] = (int)argv[1][i] -97;
            }
            else{
                shift[i]=(int)argv[1][i] - 65;
            }
        }

    string a = get_string("plaintext: ");
    printf("ciphertext: ");
    int n = sizeof(shift)/sizeof(shift[0]);
    int wrap = 0;
    for(int i =0; i<strlen(a); i++){
        if(isalpha(a[i])){
            int index = wrap%n;
            int quer = (int)a[i]+(int)shift[index];
            if(islower(a[i]) && quer>122){
                quer = quer-26;
            }
            else if(isupper(a[i]) && quer>90){
                quer = quer-26;
            }
            printf("%c",quer);
            wrap++;
        }
        else printf("%c", a[i]);
    }
    printf("\n");

    }
}

bool check(string g){
    for(int i = 0; i < strlen(g); i++){
        if(!isalpha(g[i])){
            return false;
        }
    }
    return true;
}

File: pset6/sentimental/vigenere.py
import sys
import cs50

def main():

    if len(sys.argv) !=2 or sys.argv[1].isalpha() == False:
        print("Usage: ./vigenere keyword")
        exit(1)
    else:
        shift = []
        for q in sys.argv[1]:
            if q.islower():
                shift.append(ord(q) -97)
            else:
                shift.append(ord(q) -65)

    print("plaintext: ", end="")
    a = cs50.get_string()
    print("ciphertext: ", end="")

    n = len(shift)
    wrap = 0

    for i in a:
        if i.isalpha():
            index = wrap%n
            quer = ord(i) + shift[index]

            if i.islower() and quer>122:
                quer = quer - 26
            elif i.isupper() and quer>90:
                quer = quer - 26
            print("{}".format(chr(quer)), end="")
            wrap+=1

        else:
            print("{}".format(i), end="")
    print()

if __name__ == "__main__":
    main()